### PR TITLE
feature for name added

### PR DESCRIPTION
**Pull request by casssi**
## Description

Model is now save with a name or an uuid if there no name set for the model.

The variable added in the communication for that is name model_id.

### CODE REVIEW

File: server/blindai_sgx/src/model_store.rs [hunk 1]
```diff
@@ -10,12 +10,11 @@ use std::{
     collections::{hash_map::Entry, HashMap},
     sync::Arc,
 };
-use uuid::Uuid;
 
 use crate::model::{InferenceModel, ModelDatumType, OnnxModel};
 
 struct InnerModelStore {
-    models_by_id: HashMap<Uuid, InferenceModel>,
+    models_by_nmid: HashMap<Option<String>, InferenceModel>,
```
Comment (cchudant) [inline server/blindai_sgx/src/model_store.rs:17]: why would you want an entry for None in the hashmap

HashMap<String, ...> would be much better

File: server/blindai_sgx/src/model.rs [hunk 2]
```diff
@@ -111,7 +110,7 @@ impl InferenceModel {
     pub fn load_model(
         mut model_data: &[u8],
         input_facts: Vec<Vec<usize>>,
-        model_id: Uuid,
+        model_id: Option<String>,
```
Comment (cchudant) [inline server/blindai_sgx/src/model.rs:113]: keep this an option, and generate the uuid here when None

File: server/blindai_sgx/src/model.rs [hunk 3]
```diff
@@ -170,7 +169,7 @@ impl InferenceModel {
     pub fn from_onnx_loaded(
         onnx: Arc<OnnxModel>,
         input_facts: Vec<Vec<usize>>,
-        model_id: Uuid,
+        model_id: Option<String>,
```
Comment (cchudant) [inline server/blindai_sgx/src/model.rs:172]: same

File: server/blindai_sgx/src/model_store.rs [hunk 4]
```diff
@@ -88,37 +87,41 @@ impl ModelStore {
...
 
-        match read_guard.models_by_id.get(&model_id) {
+        match read_guard.models_by_nmid.get(&model_nmid.clone()) {
             Some(model) => Some(fun(model)),
             None => None,
         }
     }
 
-    pub fn delete_model(&self, model_id: Uuid) -> Option<InferenceModel> {
+    pub fn delete_model(&self, model_nmid: Option<String>) -> Option<InferenceModel> {
```
Comment (cchudant) [inline server/blindai_sgx/src/model_store.rs:121]: this should be String

**Review by cchudant — CHANGES_REQUESTED** [review]
This is good! just a little comment

**Comment by JoFrost** [comment]
Closing as part of the input-output-rework branch.

**Pull request closed without merging**